src/layout/rectangles.js

```diff
@@ -58,7 +62,7 @@ export default function(layoutStrategy) {
             // filter the data to include only those remaining after collision removal
             var filteredData = filteredLayout.map(function(d) { return data[d.dataIndex]; });
 
-            var g = dataJoin(this, filteredData);
+            g = dataJoin(this, filteredData);
```

Comment:
If the filtering only affected `display`/`visibility`/`opacity` of the rectangle, then I don't think there wouldn't be a need for a second join. Do you think that would be a reasonable compromise?